base/math.jl

```diff
@@ -72,6 +77,13 @@ function clamp!(x::AbstractArray, lo, hi)
     x
 end
 
+function clamp!(x::AbstractArray, ::Type{T}) where {T}
```

Comment:
I guess I think of `fill!` as more like an initialization operation for the array (even though it isn't exactly)